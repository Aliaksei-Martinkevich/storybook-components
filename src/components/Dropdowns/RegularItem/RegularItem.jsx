import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './RegularItem.scss';
import Remark from './Remark/Remark.jsx';


const RegularItem = ({ onClick, text, className, remark, iconUri }) => (
  <div
    onClick={onClick}
    className={classNames('dropdown-regular-item', className)}
    title={text}
  >
    {iconUri && <img className="dropdown-regular-item__icon" src={iconUri} alt="" />}
    <span>{text}</span>
    {remark && <Remark className="dropdown-regular-item__remark" text={remark} />}
  </div>
  );

RegularItem.defaultProps = {
  onClick: undefined,
  className: '',
  remark: '',
  iconUri: '',
};

RegularItem.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  remark: PropTypes.string,
  iconUri: PropTypes.string,
};

export default RegularItem;
