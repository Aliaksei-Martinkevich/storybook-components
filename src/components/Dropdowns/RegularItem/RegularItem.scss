.dropdown-regular-item {
    padding: 5px;
    cursor: pointer;
}

.dropdown-regular-item:hover {
    background-color: #EEEEEE;
}
