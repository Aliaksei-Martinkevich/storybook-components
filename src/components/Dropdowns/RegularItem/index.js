import React from 'react';
import PropTypes from 'prop-types';

import './index.css';
import Remark from '../Remark';


const RegularItem = ({ onClick, text, className = '', remark = '', iconUri = '' }) => {
  const iconImageStyle = {
    backgroundImage: `url(${iconUri})`,
    paddingLeft: '25px',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: '2.5px center',
    backgroundSize: '20px',
  }

  return (
    <div 
    onClick={onClick}
    className={`dropdown-regular-item ${className}`}
    style={iconUri ? iconImageStyle : undefined}
    title={text}
    >
      {text}
    { remark ? <Remark text={remark}/> : '' }
    </div>
  )
};

RegularItem.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string,
  className: PropTypes.string,
  remark: PropTypes.string,
};

export default RegularItem;
