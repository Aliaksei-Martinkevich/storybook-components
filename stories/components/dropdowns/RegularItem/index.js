import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Components from '../../../../src/components'

const RegularItem = Components.Dropdowns.RegularItem;

export default storiesOf('Dropdown regular Item', module)
  .add('With text', () =>
    <RegularItem
      onClick={action('Item clicked')}
      text='Some text' />)
  .add('With long text', () =>
    <RegularItem
      onClick={action('Item clicked')}
      text='Some long long long long long long text' />);