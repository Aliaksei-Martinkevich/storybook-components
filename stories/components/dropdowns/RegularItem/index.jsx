import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import RegularItem from '../../../../src/components/Dropdowns/RegularItem/RegularItem.jsx';
import icon from '../../../../assets/icons/svg/loading-circle-dashes-black.svg';

export default storiesOf('Dropdown regular Item', module)
  .add('With text', () =>
    (<RegularItem
      onClick={action('Item clicked')}
      text="Some text"
    />))
  .add('With long text', () =>
    (<RegularItem
      onClick={action('Item clicked')}
      text="Some long long long long long long text"
    />))
  .add('With remark', () =>
    (<RegularItem
      onClick={action('Item clicked')}
      text="Some text"
      remark="Some remark"
    />))
  .add('With utf icon', () =>
    (<RegularItem
      onClick={action('Item clicked')}
      text="👁 Some text"
    />))
  .add('With icon', () =>
    (<RegularItem
      onClick={action('Item clicked')}
      text="Some text"
      iconUri={icon}
    />));
