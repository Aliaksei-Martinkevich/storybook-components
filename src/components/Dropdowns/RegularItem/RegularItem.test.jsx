import React from 'react';
import renderer from 'react-test-renderer';
import RegularItem from './RegularItem';
import icon from '../../../../assets/icons/svg/collapse_arrow-black.svg';

describe('Regular item', () => {
  it('Should render without fails', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const onClick = jest.fn();
    const component = renderer.create(
      <RegularItem
        className={customClassName}
        text={text}
        iconUri={icon}
        onClick={onClick}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
