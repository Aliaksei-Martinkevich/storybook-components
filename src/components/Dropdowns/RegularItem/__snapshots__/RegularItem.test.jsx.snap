// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regular item Should render without fails 1`] = `
<div
  className="dropdown-regular-item class-name-example"
  onClick={[Function]}
  title="Some text"
>
  <img
    alt=""
    className="dropdown-regular-item__icon"
    src="test-file-stub"
  />
  <span>
    Some text
  </span>
  
</div>
`;
